xbmc/video/videosync/VideoSyncCocoa.cpp

```diff
@@ -33,6 +33,28 @@
 #if defined(TARGET_DARWIN_OSX)
 #include <QuartzCore/CVDisplayLink.h>
 #include "osx/CocoaInterface.h"
+#endif
+
+void CVideoSyncCocoa::OnLostDevice()
+{
+  if (!m_displayLost)
+  {
+    m_displayLost = true;
+    m_lostEvent.Wait();
```

Comment:
OnResetDevice doesn't currently make Run() terminate though, which is the problem I was trying to address here.

The OnLostDevice stuff is probably not necessary, as you mentioned.